infra/modules/sns/main.tf

```diff
@@ -0,0 +1,29 @@
+resource "aws_sns_topic" "science-octopus-dlq-messages-topic" {
```

Comment:
It might be more consistent to drop the "science-octopus-" prefix from these resource names (not the name property itself, I get that that has to be unique globally). I couldn't see the prefix used in any of the other terraform resource names we have, only in the names of s3 buckets. Since the resource names just have to be unique in the scope of this project, adding "science-octopus-" might be a bit redundant.